lib/logstash/pipeline.rb

```diff
@@ -59,16 +59,20 @@ def ready?
   end
 
   def configure(setting, value)
-    if setting == "filter-workers" && value > 1
-      # Abort if we have any filters that aren't threadsafe
-      plugins = @filters.select { |f| !f.threadsafe? }.collect { |f| f.class.config_name }
-      if !plugins.size.zero?
-        raise LogStash::ConfigurationError, "Cannot use more than 1 filter worker because the following plugins don't work with more than one worker: #{plugins.join(", ")}"
-      end
-    end
     @settings[setting] = value
   end
 
+  def safe_filter_worker_count(desired)
+    safe_filters, unsafe_filters = @filters.partition(&:threadsafe?)
+    if unsafe_filters.any?
+      plugins = unsafe_filters.collect { |f| f.class.config_name }
+      @logger.warn("Setting filter threads to 1 because there are some filters that will not work with multiple worker threads", :filters => plugins)
```

Comment:
> Setting filter *workers* threads with

 We use the workers naming  covention in the doc, I think we should reuse it in the warning.